genesyscloud/outbound_contact_list_template/resource_genesyscloud_outbound_contact_list_template.go

```diff
@@ -0,0 +1,624 @@
+			},
+			`contactable_time_column`: {
+				Description: `A column that indicates the timezone to use for a given contact when checking contactable times.`,
+				Optional:    true,
+				Type:        schema.TypeString,
+			},
+		},
+	}
+
+	outboundContactListTemplateColumnDataTypeSpecification = &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			`column_name`: {
+				Description: `The column name of a column selected for dynamic queueing.`,
+				Required:    true,
+				Type:        schema.TypeString,
+			},
+			`column_data_type`: {
+				Description:  `The data type of the column selected for dynamic queueing (TEXT, NUMERIC or TIMESTAMP)`,
+				Optional:     true,
+				Computed:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.StringInSlice([]string{"TEXT", "NUMERIC", "TIMESTAMP"}, false),
+			},
+			`min`: {
+				Description: `The minimum length of the numeric column selected for dynamic queueing.`,
+				Optional:    true,
+				Type:        schema.TypeInt,
+			},
+			`max`: {
+				Description: `The maximum length of the numeric column selected for dynamic queueing.`,
+				Optional:    true,
+				Type:        schema.TypeInt,
+			},
+			`max_length`: {
+				Description: `The maximum length of the text column selected for dynamic queueing.`,
+				Required:    true,
+				Type:        schema.TypeInt,
+			},
+		},
+	}
+)
+
+func getAllOutboundContactListTemplates(_ context.Context, clientConfig *platformclientv2.Configuration) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
+	resources := make(resourceExporter.ResourceIDMetaMap)
+	outboundAPI := platformclientv2.NewOutboundApiWithConfig(clientConfig)
+
+	for pageNum := 1; ; pageNum++ {
+		const pageSize = 100
+		contactListTemplateConfigs, resp, getErr := outboundAPI.GetOutboundContactlisttemplates(pageSize, pageNum, true, "", "", "", "")
+		if getErr != nil {
+			return nil, util.BuildAPIDiagnosticError(resourceName, fmt.Sprintf("Failed to get page of contact list template configs error: %s", getErr), resp)
+		}
+
+		if contactListTemplateConfigs.Entities == nil || len(*contactListTemplateConfigs.Entities) == 0 {
+			break
+		}
+
+		for _, contactListTemplateConfig := range *contactListTemplateConfigs.Entities {
+			resources[*contactListTemplateConfig.Id] = &resourceExporter.ResourceMeta{Name: *contactListTemplateConfig.Name}
+		}
+	}
+
+	return resources, nil
+}
+
+func OutboundContactListTemplateExporter() *resourceExporter.ResourceExporter {
+	return &resourceExporter.ResourceExporter{
+		GetResourcesFunc: provider.GetAllWithPooledClient(getAllOutboundContactListTemplates),
+		RefAttrs: map[string]*resourceExporter.RefAttrSettings{
+			"attempt_limit_id": {RefType: "genesyscloud_outbound_attempt_limit"},
+			"division_id":      {RefType: "genesyscloud_auth_division"},
+		},
+	}
+}
+
+func ResourceOutboundContactListTemplate() *schema.Resource {
+	return &schema.Resource{
+		Description: `Genesys Cloud Outbound Contact List Template`,
```

Comment:
To stay consistent with the over all package structure, could you put the schema definitions for the exporter, resource and data source into a file `resource_genesyscloud_outbound_contact_list_template_schema.go`. You can add the `resourceName` constant to that file as well and use it wherever the resource name string appears

Reply:
I just put the `resourceName` in the init file